syntax = "proto3";

package forca;

service Forca {
  rpc JoinGame (JoinGameRequest) returns (JoinGameResponse);
  rpc StartGame (StartGameRequest) returns (StartGameResponse);
  rpc GuessLetter (GuessLetterRequest) returns (GuessLetterResponse);
  rpc GetGameState (GetGameStateRequest) returns (GetGameStateResponse);
}

message JoinGameRequest {
  string player_name = 1;
}

message JoinGameResponse {
  string player_id = 1;
  string message = 2;
  repeated string players = 3;
}

message StartGameRequest {
  string player_id = 1;
}

message StartGameResponse {
  string message = 1;
  bool success = 2;
}

message GuessLetterRequest {
  string player_id = 1;
  string letter = 2;
}

message GuessLetterResponse {
  string message = 1;
  bool correct = 2;
  bool game_over = 3;
  int32 points = 4;
}

message GetGameStateRequest {
  string player_id = 1;
}

message GetGameStateResponse {
  string current_word = 1;
  int32 attempts_left = 2;
  repeated string guessed_letters = 3;
  string current_player = 4;
  repeated PlayerScore scores = 5;
  bool game_over = 6;
  string message = 7;
}

message PlayerScore {
  string player_name = 1;
  int32 score = 2;
}
